sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -179,12 +179,17 @@ class OrcFileFormat
 
       val fs = filePath.getFileSystem(conf)
       val readerOptions = OrcFile.readerOptions(conf).filesystem(fs)
-      val requestedColIdsOrEmptyFile =
+      val (requestedColIdsOrEmptyFile, canPruneCols) =
         Utils.tryWithResource(OrcFile.createReader(filePath, readerOptions)) { reader =>
           OrcUtils.requestedColumnIds(
             isCaseSensitive, dataSchema, requiredSchema, reader, conf)
         }
 
+      if (!canPruneCols) {
```

Comment:
nit: we can simplify the code a bit
```
val resultSchemaString = if (canPruneCols) {
  OrcUtils.orcTypeDescriptionString(resultSchema)
} else {
  OrcUtils.orcTypeDescriptionString(StructType(dataSchema.fields ++ partitionSchema.fields))
}
```

Then we don't need to keep the `val actualSchema =...` and `var resultSchemaString =...` at the beginning.

Reply:
done